src/formats/transparent.h

```diff
@@ -0,0 +1,44 @@
+/*! \file
+ * \brief Transparent VGA pixel values
+ * \details Describes the transparancy index value for various engine components.
+ * \copyright MIT license.
+ * \date 2024
+ * \author Andrew Thompson
+ * \author Tuomas Virtanen
+ */
+
+#ifndef TRANSPARENT_H
+#define TRANSPARENT_H
+
+// Default's.
+// Please do not use the default definitions in the code directly. Define a new value below instead.
+// This way all different transparency values are stored in one place.
+#define DEFAULT_TRANSPARENT_INDEX 0
+#define DEFAULT_NOT_TRANSPARENT -1
+
+// All sprites have holes, these holes need to be transparent.
+#define SPRITE_TRANSPARENT_INDEX DEFAULT_TRANSPARENT_INDEX
+
+// Backgrounds have no transparent pixels and need all pixels to be rendered.
+#define BACKGROUND_TRANSPARENT_INDEX DEFAULT_NOT_TRANSPARENT
+
+// Force an opaque menu for now.
+#define MENU_TRANSPARENT_INDEX DEFAULT_NOT_TRANSPARENT
+
+// Portraits have a transparancy index that needs to offset past other items in the global palette.
+// TODO: Should really become default.
+#define PORTRAIT_TRANSPARENT_INDEX 0xD0
+
+// Portrait highlights need a special offset for transparency to not conflict with other entries in the global palette.
+// TODO: Should really become default.
+#define HIGHLIGHT_TRANSPARENT_INDEX 90
```

Comment:
This is not needed

Reply:
Are any of these actually needed? All the sprites already get these values by default, and our current code already informs the sprite on what the default is. I don't understand why we need any of these ?

Reply:
We keep coming back to this discussion :) 
The transparency index was a bit of mess, and although it slightly improved, it is still is a hack.
This change consolidates the hack by disallowing "sidesets" and modification of the transparency outside of initialization.

I don't agree with "it's already set by default". It's not, there are issues with 0xD0 for portraits. There is a -1 and 0.
From the code before, it was also hard to track down all the locations where transparency was being populated. This change puts it all in 1 place.

I also maintain transparency should be able to move around, per platform.

Reply:
How would you move the transparency around per platform ?

On the -1 value, that should be the renderers problem to decide what to do with -1. On GL renderer, it's just "no transparency", on other renderers it can mean whatever they want it to mean. 

Moving the value setting to the constructor is fine, I don't care either way.

Reply:
> How would you move the transparency around per platform ?
It would be an ifdef in the transparent.h

> On the -1 value, that should be the renderers problem to decide what to do with -1. On GL renderer, it's just "no transparency", on other renderers it can mean whatever they want it to mean.
When a sprite is currently "decompressed" it's filled with the transparent index (0 default), a platform could choose to select 8 as the transparent index or 255, as long as that value doesn't clash with any of the already present values in the sprite data.

Reply:
Why would a platform require other transparency indexes ?

Reply:
Optimization. Right now nuking index zero (for transparency) of the palette on the N64 actually requires a copy of the surrounding 3 palette entries. Putting these somewhere outside of the sprite's palette simplifies the transparency nuking, as I can nuke the surrounding palette entries with uninitialized memory and forego the read modify write.

Reply:
This seems like a valid point, and doesn't cost us all that much to support.

Reply:
Ok. I feel the reasoning is thought through. I'll resolve this one.